jib-gradle-plugin/src/test/java/com/google/cloud/tools/jib/gradle/FilesTaskTest.java

```diff
@@ -0,0 +1,137 @@
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.gradle;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.gradle.testkit.runner.BuildResult;
+import org.gradle.testkit.runner.BuildTask;
+import org.gradle.testkit.runner.TaskOutcome;
+import org.hamcrest.CoreMatchers;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+/** Tests for {@link FilesTask}. */
+public class FilesTaskTest {
+
+  @ClassRule public static final TestProject simpleTestProject = new TestProject("simple");
+
+  @ClassRule public static final TestProject multiTestProject = new TestProject("multi-service");
+
+  /**
+   * Verifies that the files task succeeded and returns the list of paths it prints out.
+   *
+   * @param project the project to run the task on
+   * @param moduleName the name of the sub-project, or {@code null} if no sub-project
+   * @return the list of paths printed by the task
+   */
+  private static List<Path> verifyTaskSuccess(TestProject project, @Nullable String moduleName) {
+    String taskName =
+        ":" + (moduleName == null ? "" : moduleName + ":") + JibPlugin.FILES_TASK_NAME;
+    BuildResult buildResult = project.build(taskName, "-q");
+    BuildTask jibTask = buildResult.task(taskName);
+    Assert.assertNotNull(jibTask);
+    Assert.assertEquals(TaskOutcome.SUCCESS, jibTask.getOutcome());
+
+    return Splitter.on(System.lineSeparator())
+        .omitEmptyStrings()
+        .splitToList(buildResult.getOutput())
+        .stream()
+        .map(Paths::get)
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Asserts that two lists contain the same paths. Required to avoid Mac's /var/ vs. /private/var/
+   * symlink issue.
+   *
+   * @param expected the expected list of paths
+   * @param actual the actual list of paths
+   * @throws IOException if checking if two files are the same fails
+   */
+  private static void assertPathListsAreEqual(List<Path> expected, List<Path> actual)
+      throws IOException {
+    Assert.assertEquals(expected.size(), actual.size());
+    for (int index = 0; index < expected.size(); index++) {
+      Assert.assertEquals(
+          expected.get(index).toFile().getCanonicalFile(),
+          actual.get(index).toFile().getCanonicalFile());
+    }
+  }
+
+  @Test
+  public void testFilesTask_singleProject() throws IOException {
+    Path projectRoot = simpleTestProject.getProjectRoot();
+    List<Path> result = verifyTaskSuccess(simpleTestProject, null);
+    List<Path> expected =
+        ImmutableList.of(
+            projectRoot.resolve("src/main/custom-extra-dir"),
+            projectRoot.resolve("build.gradle"),
+            projectRoot.resolve("src/main/java"),
+            projectRoot.resolve("src/main/resources"));
```

Comment:
Should this have the dependency jar?

Reply:
Not currently, since it's not a SNAPSHOT. Issue is open here #1027